functions/terraform-python-example/README.md

```diff
@@ -0,0 +1,280 @@
+
+In this tutorial you will discover an example of Instance automation using Python and Terraform: 
+
+*  Automatically shut down / start instances
+
+
+## Requirements
+
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [generated an API key](/console/my-project/how-to/generate-api-key/)
+  - You have [Python](https://www.python.org/) installed on your machine
+  - You have [Terraform](https://registry.terraform.io/providers/scaleway/scaleway/latest/docs) installed on your machine
+
+
+
+## Context
+In this tutorial we will simulate a project with a production environment that will be running all the time and a development environment that will be turn off on week-ends to save costs.
+
+## Initialize your Terraform project
+1. Create a 'Terraform' folder to store your configuration as explain in the terraform documentation.
+2. Create 5 files to configure your infrastructure:
+  a. 'main.tf': will contain the main set of configurations for your project. Here, it will be our instance
+  b. 'provider.tf': Terraform relies on plugins called “providers” to interact with remote systems
+  c. 'backend.tf': each Terraform configuration can specify a backend, which defines where the state file of the current infrastructure will be stored. Thanks to this file, Terraform keeps track of the managed resources. This state can be stored locally or remotely. Configuring a remote backend allows multiple people to work on the same infrastructure
+  d. 'variables.tf': will contain the variable definitions for your project. Since all Terraform values must be defined, any variables that are not given a default value will become required arguments
+  e. 'terraform.tfvars': allows you to set the actual value of the variables
+3. Create the following folder:
+  a. 'function': to store your function code
+  b. 'files': to temporary store your zip function code
+
+Your folder should now look like:
+```bash
+Terraform
+ | -- files
+ | -- function
+ -- main.tf
+ -- backend.tf
+ -- provider.tf
+ -- terraform.tfvars
+ -- variables.tf
+```
+
+4. Edit the 'backend.tf' file to enable distant configuration backup
+```json
```

Comment:
syntax highlighting should be `hcl` instead of `json` (everywhere in this file)

Reply:
Done!